test/NuGet.Core.Tests/NuGet.Commands.Test/PushCommandTests.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Threading.Tasks;
+using NuGet.Commands;
+using NuGet.Configuration;
+using NuGet.Frameworks;
+using NuGet.Logging;
+using NuGet.ProjectModel;
+using NuGet.Test.Utility;
+using Xunit;
+
+namespace NuGet.Commands.Test
+{
+    public class PushCommandTests
+    {
+        [Fact]
+        public async Task PushCommand_AbsolutePathSource()
+        {
+            using (TestDirectory workingDir = TestFileSystemUtility.CreateRandomTestFolder())
+            {
+                // Arrange (create a test package)
+                DirectoryInfo packageDir = new DirectoryInfo(Path.Combine(workingDir, "package"));
+                DirectoryInfo packagePushDest = new DirectoryInfo(Path.Combine(workingDir, "packagePushDest"));
+                packageDir.Create();
+                packagePushDest.Create();
+
+                List<PackageSource> packageSources = new List<PackageSource>();
+                packageSources.Add(new PackageSource(packagePushDest.FullName));
+
+                string packageFile = CreateTestPackage(packageDir.FullName);
+                string destFile = Path.Combine(packagePushDest.FullName, Path.GetFileName(packageFile));
+
+                // Act
+                await PushRunner.Run(
+                    Settings.LoadDefaultSettings(null, null, null),
+                    new TestPackageSourceProvider(packageSources),
+                    packageFile,
+                    packagePushDest.FullName,
+                    null, // api key
+                    0, // timeout
+                    false, // disable buffering
+                    false, // no symbols
+                    new TestLogger());
+
+                // Assert
+                Assert.Equal(true, File.Exists(destFile));
+            }
+        }
+
+        private static string CreateTestPackage(string packageDir, string packageId = "test", string version = "1.0.0")
+        {
+            string file = Path.Combine(packageDir, $"{packageId}.{version}.nupkg");
```

Comment:
This is fine, but it could also be done with a SimpleTestPackageContext which does very similar things